api/types/accesslist/review.go

```diff
@@ -58,6 +69,20 @@ type ReviewSpec struct {
 	Changes ReviewChanges `json:"changes" yaml:"changes"`
 }
 
+// Clone returns a copy of the review spec.
+func (r *ReviewSpec) Clone() ReviewSpec {
+	if r == nil {
+		return ReviewSpec{}
+	}
+	return ReviewSpec{
+		AccessList: r.AccessList,
+		Reviewers:  slices.Clone(r.Reviewers),
```

Comment:
Here’s a silly example, but good enough to illustrate the point. 

Imagine if we ever changed r.Reviewers to []*string: in that case, this line would compile fine and the test would still pass. But the clone wouldn’t actually be independent, because we’d still be copying the string pointers, not the string values, meaning both objects would reference the same underlying data and if we change them, we add data-races 

Hand-writing this code is basically asking for trouble. People can still make mistakes without realizing. Same if we ever change from []string to []*ReviewersXYZ

Reply:
As I said I want to keep the discussion in https://github.com/gravitational/teleport/pull/58588 if we agree that we want to use deepCopy it will take me   5 min to rework this logic.

Reply:
All Right, Since i have approvals on https://github.com/gravitational/teleport/pull/58588 and can merge this I will rework the approuch to deepcopy.